src/textual/widgets/_tabs.py

```diff
@@ -0,0 +1,440 @@
+    }
+    Tabs #tabs-list {
+       width: auto;
+       min-height: 2;
+    }
+    Tabs #tabs-list-bar, Tabs #tabs-list {
+        width: auto;
+        height: auto;
+        min-width: 100%;
+        overflow: hidden hidden;
+    }
+    Tabs:focus .underline--bar {
+        background: $foreground 20%;
+    }
+    """
+
+    BINDINGS: ClassVar[list[BindingType]] = [
+        Binding("left", "previous_tab", "Previous tab", show=False),
+        Binding("right", "next_tab", "Next tab", show=False),
+    ]
+
+    class TabActivated(Message):
+        """Sent when a new tab is activated."""
+
+        def __init__(self, tab: Tab | None) -> None:
+            self.tab = tab
+            super().__init__()
+
+        def __rich_repr__(self) -> rich.repr.Result:
+            yield self.tab
+
+    active: reactive[str] = reactive(str, init=False)
+
+    def __init__(
+        self,
+        *tabs: Tab | TextType,
+        active: str | None = None,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Construct a Tabs widget.
+
+        Args:
+            *tabs: Positional argument should be explicit Tab objects, or a str or Text.
+            active (_type_, optional): ID of the tab which should be active on start.
+            name: Optional name for the input widget.
+            id: Optional ID for the widget.
+            classes: Optional initial classes for the widget.
+            disabled: Whether the input is disabled or not.
+        """
+        self._tabs_counter = 0
+        add_tabs = [
+            (
+                Tab(tab, id=f"tab-{self._new_tab_id}")
+                if isinstance(tab, (str, Text))
+                else tab
+            )
+            for tab in tabs
+        ]
+        super().__init__(
+            name=name,
+            id=id,
+            classes=classes,
+            disabled=disabled,
+        )
+        self._tabs = add_tabs
+        self._first_active = active
+
+    @property
+    def _new_tab_id(self) -> int:
+        """Get the next tab id in a sequence."""
+        self._tabs_counter += 1
+        return self._tabs_counter
+
+    @property
+    def tab_count(self) -> int:
+        """The total number of tabs."""
+        return len(self.query("#tabs-list > Tab"))
+
+    @property
+    def _next_active(self) -> Tab | None:
+        """Get the next tab to make active, if the active tab is removed."""
+        tabs = list(self.query("#tabs-list > Tab").results(Tab))
+        if self.active_tab is None:
+            return None
+        try:
+            active_index = tabs.index(self.active_tab)
+        except ValueError:
+            return None
+        del tabs[active_index]
+        try:
+            return tabs[active_index]
+        except IndexError:
+            try:
+                return tabs[active_index - 1]
+            except IndexError:
+                pass
+        return None
```

Comment:
There is something I don't get. Can't we just do this by inspecting the length of `tabs` with respect to the `active_index`?
Why do we have to `del tabs[active_index]` and whatnot?

Does `del tabs[active_index]` do anything other than deleting that specific entry from the list `tabs`?

Reply:
I'm sure there is a more elegant solution. But we want to get the next active tab, which is normally the one to the right. Unless we're at the end, then its the one to the left.